lib/rack/handle_bad_encoding.rb

```diff
@@ -5,11 +5,11 @@ def initialize(app)
     end
 
     def call(env)
-      if %w[/location-suggestions /provider-suggestions].include?(env["REQUEST_PATH"])
+      if %w[/location-suggestions /provider-suggestions /results].include?(env["REQUEST_PATH"])
         begin
           Rack::Utils.parse_nested_query(env["QUERY_STRING"].to_s)
         rescue Rack::Utils::InvalidParameterError
-          env["QUERY_STRING"] = ""
+          return [301, { "Location" => "/404", "Content-Type" => "text/html" }, []]
```

Comment:
This looks like 301 redirect to the /404 page, does that mean that a user would "lose" the page which caused the error once they hit it? Would that be annoying to deal with?

Reply:
Based on the errors we have seen on Sentry most of these types of invalid queries result from:

1. Users copy and pasting urls containing invalid param (e.g non-ASCII characters). 
2. Bots or malicious users playing around with params

So, in these cases there wasn't a previous page to lose ...

But yes if a user somehow managed to create an invalid query via a form (i.e. entering an invalid postcode, address or provider here https://www.qa.find-postgraduate-teacher-training.service.gov.uk/) then they would lose the page but 
have not seen evidence of this happening (yet).